src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/PledgeService.cs

```diff
@@ -1,29 +1,34 @@
 ﻿using Newtonsoft.Json;
 using Newtonsoft.Json.Converters;
+using SFA.DAS.Encoding;
 using SFA.DAS.LevyTransferMatching.Infrastructure.Dto;
-using SFA.DAS.LevyTransferMatching.Infrastructure.Services.PledgeService;
-using System;
-using System.Collections.Generic;
 using System.Net.Http;
-using System.Text;
 using System.Threading.Tasks;
 
 namespace SFA.DAS.LevyTransferMatching.Infrastructure.Services.PledgeService
 {
     public class PledgeService : IPledgeService
     {
         private readonly HttpClient _client;
+        private readonly IEncodingService _encodingService;
 
-        public PledgeService(HttpClient client)
+        public PledgeService(HttpClient client, IEncodingService encodingService)
         {
             _client = client;
+            _encodingService = encodingService;
         }
 
-        public async Task PostPledge(PledgeDto pledgeDto, long accountId)
+        public async Task<PledgeDto> PostPledge(PledgeDto pledgeDto, long accountId)
         {
             var json = JsonConvert.SerializeObject(pledgeDto, new StringEnumConverter());
-            var response = await _client.PostAsync($"accounts/{accountId}/pledges", new StringContent(json, Encoding.UTF8, "application/json"));
+            var response = await _client.PostAsync($"accounts/{accountId}/pledges", new StringContent(json, System.Text.Encoding.UTF8, "application/json"));
             response.EnsureSuccessStatusCode();
+
+            var result = JsonConvert.DeserializeObject<PledgeDto>(await response.Content.ReadAsStringAsync());
+
+            pledgeDto.Id = _encodingService.Encode(long.Parse(result.Id), EncodingType.PledgeId);
```

Comment:
You're doing the encoding within the service - I'd have probably put it into the orchestrator, but I'm open to either.

However, I'd have just returned the Id (encoded or otherwise) from this method, rather than an object. The object returned is actually the one passed in, except that the Id property is set. It feels wrong to modify a property of a passed-in object.

Reply:
Will change to just return the encoded id

My thoughts on where to encode/decode was driven by which representation the website will likely require and it seemed that it would be the encoded pledge id rather than the raw id so putting it in the pledge service seemed like the better option?

Reply:
Changed to encode in the Orchestrator